server/app/views/components/LinkElement.java

```diff
@@ -114,15 +122,21 @@ public LinkElement opensInNewTab() {
     return this;
   }
 
-  public LinkElement setIcon(Icons icon) {
+  public LinkElement setIcon(Icons icon, IconPosition position) {
     this.icon = Optional.of(checkNotNull(icon));
+    this.iconPosition = position;
     return this;
   }
 
   public ATag asAnchorText() {
     ATag tag = a();
-    this.icon.ifPresent(icon -> tag.with(Icons.svg(icon).withClasses("mr-2", "w-5", "h-5")));
-    tag.withText(this.text);
+    if (this.iconPosition == IconPosition.END) {
```

Comment:
suggestion (blocking): per java style guide
```suggestion
    if (IconPosition.END.equals(this.iconPosition)) {
```